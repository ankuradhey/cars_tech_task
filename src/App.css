.flex-md-row {
    flex-direction: row;
}
.align-items-center {
    align-items: center;
}
.flex-column {
    flex-direction: column;
}
.d-flex {
    display: flex !important;
}
.brand {
    height: 48px;
}
.brand img {
    height: 48px;
}
