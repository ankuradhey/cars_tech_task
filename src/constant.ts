export const constant = {
    host: "https://auto1-mock-server.herokuapp.com",
    carsUrl: "/api/cars",
    colorsUrl: "/api/colors",
    manufacturersUrl: "/api/manufacturers",
    carDetailsUrl: "/api/cars/{stockNumber}",
};
