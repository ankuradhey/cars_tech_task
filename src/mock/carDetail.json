{
    "car": {
        "stockNumber": 10034,
        "manufacturerName": "Fiat",
        "modelName": "Grande Punto",
        "color": "black",
        "mileage": {
            "number": 170648,
            "unit": "km"
        },
        "fuelType": "Diesel",
        "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
    }
}