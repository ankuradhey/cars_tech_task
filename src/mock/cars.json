{
    "cars": [
        {
            "stockNumber": 10019,
            "manufacturerName": "Fiat",
            "modelName": "Idea",
            "color": "blue",
            "mileage": {
                "number": 111955,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10084,
            "manufacturerName": "Skoda",
            "modelName": "Rapid",
            "color": "white",
            "mileage": {
                "number": 166819,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10176,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "Sprinter",
            "color": "silver",
            "mileage": {
                "number": 125331,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10217,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "Viano",
            "color": "green",
            "mileage": {
                "number": 119173,
                "unit": "km"
            },
            "fuelType": "Diesel",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10239,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "SL-Klasse",
            "color": "blue",
            "mileage": {
                "number": 165176,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10283,
            "manufacturerName": "Chrysler",
            "modelName": "Crossfire",
            "color": "black",
            "mileage": {
                "number": 114177,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10385,
            "manufacturerName": "Porsche",
            "modelName": "718 Cayman",
            "color": "white",
            "mileage": {
                "number": 180397,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10477,
            "manufacturerName": "BMW",
            "modelName": "i3",
            "color": "blue",
            "mileage": {
                "number": 127584,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10481,
            "manufacturerName": "Volkswagen",
            "modelName": "Crafter",
            "color": "white",
            "mileage": {
                "number": 176629,
                "unit": "km"
            },
            "fuelType": "Diesel",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10571,
            "manufacturerName": "Dodge",
            "modelName": "Nitro",
            "color": "yellow",
            "mileage": {
                "number": 115196,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        }
    ],
    "totalPageCount": 100,
    "totalCarsCount": 1000
}