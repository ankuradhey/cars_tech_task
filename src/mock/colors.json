{"colors":["red","blue","green","black","yellow","white","silver"]}