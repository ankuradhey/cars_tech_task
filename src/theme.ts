import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
