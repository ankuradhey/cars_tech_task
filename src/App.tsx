import React from "react";
import { Route, Switch } from "react-router-dom";
import { Header } from "./components/Header";
import { Footer } from "./components/Footer";
import "./App.css";
import { Home } from "./components/Home";
import { CarDetails } from "./components/CarDetails/CarDetails";
import { NotFound } from "./components/NotFound";

function App() {
    return (
        <>
            <Header />
            <Switch>
                <Route exact path="/" component={Home} />
                <Route path="/car/:id" component={CarDetails} />
                <Route component={NotFound} />
            </Switch>
            <Footer />
        </>
    );
}

export default App;
