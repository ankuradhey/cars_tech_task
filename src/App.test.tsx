import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { MemoryRouter } from "react-router-dom";
import nock from "nock";
import { defaultTheme } from "./theme";
import App from "./App";
import { ThemeProvider } from "styled-components";
import { constant } from "./constant";
import * as carsMock from "./mock/cars.json";
import * as carDetailMock from "./mock/carDetail.json";
import * as colorsMock from "./mock/colors.json";
import * as manufacturersMock from "./mock/manufacturers.json";

describe("App Component", () => {
    let history;
    beforeEach(() => {
        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(`${constant.carsUrl}?page=1`)
            .reply(200, carsMock);

        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(constant.colorsUrl)
            .reply(200, colorsMock);

        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(constant.manufacturersUrl)
            .reply(200, manufacturersMock);

        history = createMemoryHistory();
    });

    it("Should do full app rendering/navigating", () => {
        const { container } = render(
            <ThemeProvider theme={defaultTheme}>
                <MemoryRouter initialEntries={["/"]}>
                    <App />
                </MemoryRouter>
            </ThemeProvider>
        );
        expect(screen.getByText(/Available cars/i)).toBeInTheDocument();
        expect(screen.getByText(/AUTO1 Group 2018/i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("Should be landing on a bad page", () => {
        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <MemoryRouter initialEntries={["/unknown"]}>
                    <App />
                </MemoryRouter>
            </ThemeProvider>
        );
        expect(container).toMatchSnapshot();
        expect(getByText(/404 - Not Found/i)).toBeInTheDocument();
        expect(
            getByText(/Sorry, the page you are looking for does not exist./i)
        ).toBeInTheDocument();
    });

    it("Should be landing on a car details", async () => {
        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(`${constant.carDetailsUrl.replace("{stockNumber}", "10058")}`)
            .reply(200, carDetailMock);

        history.push({ pathname: "/car/10058" });
        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <MemoryRouter initialEntries={["/car/10058"]}>
                    <App />
                </MemoryRouter>
            </ThemeProvider>
        );
        expect(getByText("Loading...")).toBeInTheDocument();
        await waitFor(() =>
            screen.findByText(
                `${carDetailMock.car.manufacturerName} ${carDetailMock.car.modelName}`
            )
        );
        expect(container).toMatchSnapshot();
        expect(
            getByText(`${carDetailMock.car.manufacturerName} ${carDetailMock.car.modelName}`)
        ).toBeInTheDocument();
    });
});
