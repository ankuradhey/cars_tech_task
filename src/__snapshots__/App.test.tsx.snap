// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should be landing on a bad page 1`] = `
<div>
  <div
    class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow"
  >
    <a
      class="brand my-0 mr-md-auto font-weight-normal"
      href="/"
    >
      <img
        alt="Auto1.com"
        src="logo.png"
      />
    </a>
    <nav
      class="my-2 my-md-0 mr-md-3"
    >
      <a
        class="p-2 text-dark"
        href="/"
      >
        Purchase
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        My Orders
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        Sell
      </a>
    </nav>
  </div>
  <div
    class="d-flex flex-column justify-content-center align-items-center container"
    style="min-height: calc(100vh - 160px);"
  >
    <img
      alt="logo"
      class="margin-bottom-3 d-flex"
      src="logo.png"
      width="190px"
    />
    <div
      class="sc-hKgILt hXCUdi margin-bottom-3 d-flex"
    >
      404 - Not Found
    </div>
    <div
      class="sc-hKgILt kOrGAu margin-bottom-3 d-flex"
    >
      Sorry, the page you are looking for does not exist.
    </div>
    <div
      class="sc-hKgILt kOrGAu margin-bottom-2 d-flex"
    >
      You can always go back to the  
      <button
        class="sc-dlfnbm fyhdMX"
      >
         homepage
      </button>
    </div>
  </div>
  <footer
    class="sc-bdfBwQ hhKrKQ footer text-center mt-3"
  >
    <span
      class="margin-auto"
    >
      © AUTO1 Group 2018
    </span>
  </footer>
</div>
`;

exports[`App Component Should be landing on a car details 1`] = `
<div>
  <div
    class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow"
  >
    <a
      class="brand my-0 mr-md-auto font-weight-normal"
      href="/"
    >
      <img
        alt="Auto1.com"
        src="logo.png"
      />
    </a>
    <nav
      class="my-2 my-md-0 mr-md-3"
    >
      <a
        class="p-2 text-dark"
        href="/"
      >
        Purchase
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        My Orders
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        Sell
      </a>
    </nav>
  </div>
  <div
    class="margin-bottom-3 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="sc-gsTCUz fbaEUS"
          height="400px"
          loading="false"
          width="100%"
        >
          <img
            alt="car"
            src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
    style="width: 800px;"
  >
    <div
      class="row"
    >
      <div
        class="col-md-7"
      >
        <div
          class="sc-hKgILt hXCUdi d-block margin-bottom-3"
        >
          Fiat
           
          Grande Punto
        </div>
        <div
          class="sc-hKgILt kOrGAu d-block margin-bottom-3"
        >
          Stock # 
          10034
           - 
          170648
           
          KM
           
          - 
          Diesel
           - 
          black
        </div>
        <div
          class="sc-hKgILt leLkki d-block margin-bottom-3 text-justify"
        >
          This car is currently available and can be delivered as sson as tomorrow morning. Please be aware that delivery times shown in this page are not definitive and may change due to bad weather conditions.
        </div>
      </div>
      <div
        class="col-md-5"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="sc-hKgILt leLkki d-block margin-bottom-1 text-justify"
            >
              If you like this car, click the button and save it in your collection of favorite items.
            </div>
            <button
              class="sc-iBPRYJ ptxhL float-right"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="sc-bdfBwQ hhKrKQ footer text-center mt-3"
  >
    <span
      class="margin-auto"
    >
      © AUTO1 Group 2018
    </span>
  </footer>
</div>
`;

exports[`App Component Should do full app rendering/navigating 1`] = `
<div>
  <div
    class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow"
  >
    <a
      class="brand my-0 mr-md-auto font-weight-normal"
      href="/"
    >
      <img
        alt="Auto1.com"
        src="logo.png"
      />
    </a>
    <nav
      class="my-2 my-md-0 mr-md-3"
    >
      <a
        class="p-2 text-dark"
        href="/"
      >
        Purchase
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        My Orders
      </a>
      <a
        class="p-2 text-dark"
        href="/"
      >
        Sell
      </a>
    </nav>
  </div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="margin-bottom-3 card"
        >
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  <div
                    class="sc-hKgILt llewCg"
                  >
                    Color
                  </div>
                </label>
                <select
                  class="custom-select"
                  data-testid="color"
                >
                  <option
                    value=""
                  >
                    All car colors
                  </option>
                </select>
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  <div
                    class="sc-hKgILt llewCg"
                  >
                    Manufacturer
                  </div>
                </label>
                <select
                  class="custom-select"
                  data-testid="manufacturer"
                >
                  <option
                    value=""
                  >
                    All manufacturers
                  </option>
                </select>
              </div>
            </form>
            <button
              class="sc-iBPRYJ ptxhL float-right"
              data-testid="filter"
            >
              Filter
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-md-8"
      >
        <div
          class="sc-hKgILt BnbLY d-block margin-bottom-2"
        >
          Available cars
        </div>
        <div
          class="sc-hKgILt kOrGAu margin-bottom-3"
        >
          Loading
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-2 card"
        >
          <div
            class="sc-eCssSg eGHMQQ"
          >
            <div
              class="sc-jSgupP iYzSIw"
            >
              <div
                class="sc-gsTCUz hpwOXo"
                height="80px"
                loading="true"
                width="100px"
              />
            </div>
            <div
              class="sc-gKsewC bmUAys"
            >
              <div
                class="sc-gsTCUz lfxAIM margin-bottom-1"
                height="18px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz ekjGVy margin-bottom-2"
                height="12px"
                loading="true"
                width="100%"
              />
              <div
                class="sc-gsTCUz dXSIdd margin-bottom-2 d-inline-block"
                height="12px"
                loading="true"
                width="auto"
              >
                <button
                  class="sc-dlfnbm fyhdMX invisible"
                >
                  View details
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="p-2 bd-highlight"
          >
            <div
              class="sc-hKgILt llewCg"
              data-testid="first-page"
            >
              First
            </div>
          </div>
          <div
            class="p-2 bd-highlight"
          >
            <div
              class="sc-hKgILt llewCg"
              data-testid="previous-page"
            >
              Previous
            </div>
          </div>
          <div
            class="p-2 bd-highlight"
          >
            <div
              class="sc-hKgILt llewCg"
            >
              Page 
              1
               of 
            </div>
          </div>
          <div
            class="p-2 bd-highlight"
          >
            <div
              class="sc-hKgILt llewCg"
              data-testid="next-page"
            >
              Next
            </div>
          </div>
          <div
            class="p-2 bd-highlight"
          >
            <div
              class="sc-hKgILt llewCg"
              data-testid="last-page"
            >
              Last
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="sc-bdfBwQ hhKrKQ footer text-center mt-3"
  >
    <span
      class="margin-auto"
    >
      © AUTO1 Group 2018
    </span>
  </footer>
</div>
`;
